Document/0x05b-Basic-Security_Testing.md

```diff
@@ -714,11 +714,43 @@ Other resources where you might find useful information are:
 
 ##### Xposed
 
--- ToDo: <https://github.com/OWASP/owasp-mstg/issues/1234>
-
 [Xposed](http://repo.xposed.info/module/de.robv.android.xposed.installer) is a "framework for modules that can change the behavior of the system and apps without touching any APKs." Technically, it is an extended version of Zygote that exports APIs for running Java code when a new process is started. Running Java code in the context of the newly instantiated app makes it possible to resolve, hook, and override Java methods belonging to the app. Xposed uses [reflection](https://docs.oracle.com/javase/tutorial/reflect/ "Reflection Tutorial") to examine and modify the running app. Changes are applied in memory and persist only during the process' run times—no patches to the application files are made.
 
-To use Xposed, you need to first install the Xposed framework on a rooted device. Deploy modifications deployed in the form of separate apps ("modules"), which can be toggled on and off in the Xposed GUI.
+To use Xposed, you need to first install the Xposed framework on a rooted device as explained on [XDA-Developers XPosed framework hub](https://www.xda-developers.com/xposed-framework-hub/). Deploy modifications deployed in the form of separate apps ("modules"), which can be toggled on and off in the Xposed GUI.
+
+Note: given that a plain installation of the Xposed framework is easily detected with Safetynet, we recommend using Magisk to install Xposed. This way, applications with Safetynet attestation should hav a higher chance of being testable with Xposed modules.
+
+One advantage of using Xposed to in comparison with Frida, is that the application does not have to be repackaged and resigned. Instead, you can manipulate an app directly on the device. The disadvantage is ofcourse, that you have to root the device on which you test it. This can be done on an emulator as well. See the script below for an example:
```

Comment:
Xposed to in -> Xposed in

Reply:
Frida also doesn't need repackaged applications in the same conditions as Xposed (rooted device), so that's not a valid argument imo.

A valid argument can be "Some applications may behave weirdly or crash when instrumented with Frida and Xposed can be a good backup option".

Additionally, there are many open source modules available for XPosed, so there are a lot of examples available. You also write directly in Java so there is no real intermediate layer. That makes the learning curve easier than Frida (imo again).

Reply:
> This can be done on an emulator as well. See the script below for an example:

to

> XPosed can also be installed on an emulator through the following script:

Reply:
I guess the valid arguments to me are:
- when there is basic frida detection, Xposed can help.
- when you want to have easily alterable modules, Xposed or Frida can be used: it depends on what you like the most.